src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetLegalNameHandler.cs

```diff
@@ -1,81 +1,125 @@
-            textListing.DataRows.Add(new TabularDataRow { Columns = new List<string> { $"Application submitted on: {applicationSubmittedOn}" } });
-            textListing.DataRows.Add(new TabularDataRow { Columns = new List<string> { $"Sources checked on: {applicationSourcesCheckedOn}" } });
-            model.TextListing = textListing;
-
-            // building the tables
-            var table = new TabularData { DataRows = new List<TabularDataRow>(), HeadingTitles = new List<string> { "Source", "Legal name" } };
-
-            table.DataRows.Add(new TabularDataRow { Columns = new List<string> { "Submitted application data", submittedApplicationData } });
-            table.DataRows.Add(new TabularDataRow { Columns = new List<string> { "UKRLP data", ukrlpData } });
-
-
-            // the following 2 will be dynamic, depending on whether its a company or charity
-            // these two depend on company etc
-            var companiesHouseData = "CompaniesHouse: LegalName";
-            table.DataRows.Add(new TabularDataRow { Columns = new List<string> { "Companies House data", companiesHouseData } });
-
-            var charityCommissionData = "CharityCommission: LegalName";
-            table.DataRows.Add(new TabularDataRow { Columns = new List<string> { "Charity Commission data", charityCommissionData } });
-
-            model.Tables.Add(table);
+        public async Task<LegalNamePageViewModel> Handle(GetLegalNameRequest request, CancellationToken cancellationToken)
 
-            // write this model straight to the database, and then return it
+        {
+            var pageId = "1-10";
+
+            var model = new LegalNamePageViewModel { ApplicationId = request.ApplicationId, PageId = pageId};
+ 
+            var currentRecord = await _applyApiClient.GetGatewayPageAnswer(request.ApplicationId, pageId);
+            var applicationDetails = await _applyApiClient.GetApplication(model.ApplicationId);
+
+            var gatewayReviewStatus = string.Empty;
+            if (applicationDetails?.GatewayReviewStatus != null)
+            {
+                gatewayReviewStatus = applicationDetails.GatewayReviewStatus;
+            }
+
+            if (currentRecord?.GatewayPageData != null)
+            {
+                model = JsonConvert.DeserializeObject<LegalNamePageViewModel>(currentRecord.GatewayPageData);
+                model.Status = currentRecord.Status;
+                model.GatewayReviewStatus = gatewayReviewStatus;
+                return model;
+            }
+
+            model.GatewayReviewStatus = gatewayReviewStatus;
+            if (applicationDetails?.ApplyData?.ApplyDetails?.ApplicationSubmittedOn != null)
+                model.ApplicationSubmittedOn = applicationDetails.ApplyData.ApplyDetails.ApplicationSubmittedOn;
+
+            model.SourcesCheckedOn = DateTime.Now;
+
+            var qnaApplyData = await _qnaApiClient.GetApplicationData(model.ApplicationId);
```

Comment:
You could use the QnA API endpoint {applicationId}/applicationData/{questionTag} here, rather than fetching all of the application data and parsing it

Reply:
I had to get the application data just to get the ApplicationSubmittedOn value, so thought I'd just reuse everything fetched anyway....